### PR TITLE
Integrate REST API into frontend

### LINKED ISSUE #45: Integrate API into frontend pages
- [x] Integrate into HomePage
- [x] Integrate into NicknamePage
- [x] Add input validation for game code and nickname
- [x] Add styling for error states
- [x] Add documentation for running full app

### PR DESCRIPTION
**Pull request by RawiriHohepa**
Closes #45 

- Call /game/validate and /game/create from HomePage
- Call /game/validate (on load) and /player/create from NicknamePage
- Check that game code is a number before calling /game/validate
- Add documentation for starting frontend and backend together (now required for full functionality)

### CODE REVIEW

File: frontend/src/pages/NicknamePage/index.tsx [hunk 1]
```diff
@@ -1,13 +1,60 @@
...
+const useUserIdState = createPersistedState("userId");
+
+type Props = {
+  gameCode: string;
+};
 
+const NicknamePage = ({ gameCode }: Props) => {
   const [nickname, setNickname] = useState("");
+  const [error, setError] = useState("");
+  const [userId, setUserId] = useUserIdState("");
```
Comment (taitfuller) [inline frontend/src/pages/NicknamePage/index.tsx:20]: I'm running into an error thrown from here testing locally - seems to be an issue with persisted state

File: frontend/src/pages/HomePage/index.tsx [hunk 2]
```diff
@@ -3,12 +3,40 @@ import { Link, useHistory } from "react-router-dom";
...
+    const res = await createGame();
+
+    if (res.success) {
+      if (res.data) {
+        browserHistory.push(`/${res.data}`);
+      } else {
+        setNewGameError("Unknown Error, please try again");
+      }
+    } else {
+      setNewGameError(res.error);
```
Comment (taitfuller) [inline frontend/src/pages/HomePage/index.tsx:36]: This returns the following error:

image

Comment (RawiriHohepa) [reply frontend/src/pages/HomePage/index.tsx:36]: Ah yeah this is a problem with createGame(), it's returning an object as res.error, which this then tries to render

Comment (taitfuller) [reply frontend/src/pages/HomePage/index.tsx:36]: Interesting - is this expected behaviour @Alxjlv?

Comment (RawiriHohepa) [reply frontend/src/pages/HomePage/index.tsx:36]: I should just be able to return a generic server error message, I don't think there's anything useful in the XMLHttpRequest object to return to the user

Comment (RawiriHohepa) [reply frontend/src/pages/HomePage/index.tsx:36]: Do you still get the error now @taitfuller?

File: frontend/src/pages/NicknamePage/index.tsx (hunk 1 above)
Comment (taitfuller) [reply frontend/src/pages/NicknamePage/index.tsx]: This appears to be resolved now! 🎉

File: frontend/src/pages/HomePage/index.tsx (hunk 2 above)
Comment (taitfuller) [reply frontend/src/pages/HomePage/index.tsx:36]: This is fixed now - great work!

**Review by taitfuller — APPROVED** [review]
Great work man!

**Pull request merged**